src/System.Diagnostics.TraceSource/src/System/Diagnostics/TraceInternal.cs

```diff
@@ -49,9 +63,6 @@ public static TraceListenerCollection Listeners
                             defaultListener.IndentLevel = t_indentLevel;
                             defaultListener.IndentSize = s_indentSize;
                             s_listeners.Add(defaultListener);
-                            // This is where we override default DebugProvider because we know
-                            // for sure that we have some Listeners to write to.
-                            Debug.SetProvider(s_provider);
```

Comment:
Prior to this PR, we would only activate Listeners wiring to Debug if Trace.Listeners was getting called.

Now after moving `Debug.SetProvider` to static constructor of TraceInternal, even as soon as Trace.IndentLevel (or Trace.IndentSize) get called then we can be sure values of IndentSize and IndentLevel remain in sync for both Trace and Debug. 
(This makes .NET Core behavior more consistent with Desktop)